profiles/mojoshader_profile.h

```diff
@@ -0,0 +1,365 @@
+
+typedef struct Profile
+{
+    const char *name;
+    emit_start start_emitter;
+    emit_end end_emitter;
+    emit_phase phase_emitter;
+    emit_global global_emitter;
+    emit_array array_emitter;
+    emit_const_array const_array_emitter;
+    emit_uniform uniform_emitter;
+    emit_sampler sampler_emitter;
+    emit_attribute attribute_emitter;
+    emit_finalize finalize_emitter;
+    varname_function get_varname;
+    const_array_varname_function get_const_array_varname;
+} Profile;
+
+// Common utilities...
+
+void out_of_memory(Context *ctx);
+void *Malloc(Context *ctx, const size_t len);
+char *StrDup(Context *ctx, const char *str);
+void Free(Context *ctx, void *ptr);
+void * MOJOSHADERCALL MallocBridge(int bytes, void *data);
+void MOJOSHADERCALL FreeBridge(void *ptr, void *data);
+
+int set_output(Context *ctx, Buffer **section);
+void push_output(Context *ctx, Buffer **section);
+void pop_output(Context *ctx);
+
+uint32 ver_ui32(const uint8 major, const uint8 minor);
+int shader_version_supported(const uint8 maj, const uint8 min);
+int shader_version_atleast(const Context *ctx, const uint8 maj,
+                           const uint8 min);
+int shader_version_exactly(const Context *ctx, const uint8 maj,
+                           const uint8 min);
+int shader_is_pixel(const Context *ctx);
+int shader_is_vertex(const Context *ctx);
+
+int isfail(const Context *ctx);
+void failf(Context *ctx, const char *fmt, ...);
+void fail(Context *ctx, const char *reason);
+
+void output_line(Context *ctx, const char *fmt, ...);
+void output_blank_line(Context *ctx);
+
+void floatstr(Context *ctx, char *buf, size_t bufsize, float f,
+              int leavedecimal);
+
+RegisterList *reglist_insert(Context *ctx, RegisterList *prev,
+                             const RegisterType regtype,
+                             const int regnum);
+RegisterList *reglist_find(const RegisterList *prev,
+                           const RegisterType rtype,
+                           const int regnum);
+RegisterList *set_used_register(Context *ctx,
+                                const RegisterType regtype,
+                                const int regnum,
+                                const int written);
+void set_defined_register(Context *ctx, const RegisterType rtype,
+                          const int regnum);
+
+int writemask_xyzw(const int writemask);
+int writemask_xyz(const int writemask);
+int writemask_xy(const int writemask);
+int writemask_x(const int writemask);
+int writemask_y(const int writemask);
+int replicate_swizzle(const int swizzle);
+int no_swizzle(const int swizzle);
+int vecsize_from_writemask(const int m);
+void set_dstarg_writemask(DestArgInfo *dst, const int mask);
+
+int isscalar(Context *ctx, const MOJOSHADER_shaderType shader_type,
+             const RegisterType rtype, const int rnum);
+
+static const char swizzle_channels[] = { 'x', 'y', 'z', 'w' };
```

Comment:
May be a possible cause of increased size, but just barely...

Reply:
Yeah, it's annoying. I've tried moving it around and making it non-static, but no matter what it doesn't seem to affect the stripped binary size. :/

Reply:
That's hopefully the optimizer being smart. Oh well, we can just leave this...